teos/src/responder.rs

```diff
@@ -290,20 +281,33 @@ impl Responder {
                     dispute_txid
                 ),
                 x => unreachable!(
```

Comment:
The logic here is still not right. This is called during the first connected block after a reorg with all data flagged as reorged out, but the backend is in the tip after a reorg. Therefore, it could be the case that something that you are sending to the network is already confirmed, which you are not converting here.

Also, it is worth checking if something is either in out cache or mempool before trying to send it, given it will be cheaper (`sendrawtransaction` forces a tx relay to all peers, which also have privacy implications on the transaction origin). Check how the code in master is currently dealing with this.

Reply:
Confirmed ones would be covered in the `ConfirmationStatus::IrrevocablyResolved` case.
The best way to handle this would be to hold on handling the reorged txs until we reach the tip of the stronger chain (just before the tip).

> it is worth checking if something is either in out cache or mempool before trying to send it

Mempool makes sense, but cache won't work because we aren't fully synced yet as our backend.

Reply:
> but cache won't work because we aren't fully synced yet as our backend

Looking at the code in `master`, we actually have one connected block from the stronger chain in our txindex to check against. But that's not a general solution if we assume we might have 1+ block reorgs.

I will need to rework the responder in a follow up anyways (`hold on handling the reorged txs until we reach the tip` & `make a DB representation for reorged status`), let's handle this then if `ConfirmationStatus::IrrevocablyResolved` does the job for now.